mods/parsoid.js

```diff
@@ -459,7 +460,27 @@ PSP.transformRevision = function (restbase, req, from, to) {
         var body2 = {
             original: original
         };
-        body2[from] = req.body[from];
+        if (from === 'sections') {
+            try {
+                var sections = JSON.parse(req.body.sections);
```

Comment:
Two small remarks:

- You might want to check that `req.body.sections` is not already something that's not a String or Buffer before parsing it, which it will be in case the client supplies the `application/json` *Content-Type* header.
- Probably only the JSON parsing should be enclosed in a `try/catch` block, since `replaceSections` is going to throw its own errors.